File: backend/src/main/java/com/artifax_solutions/website/model/ShopNowSection.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class ShopNowSection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String subtitle;
    private String title;
    private String description;
    private String buttonText;
    private String buttonUrl;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "shop_now_section_id")
    private List<ShopNowCardSection> cardList;
}

File: frontend/src/app/models/ShopNowSection.ts
export interface ShopNowSection {
  id: number | null | undefined;
  subtitle: string | null | undefined;
  title: string | null | undefined;
  description: string | null | undefined;
  buttonText: string | null | undefined;
  buttonUrl: string | null | undefined;
  cardList: ShopNowCardSection[] | undefined | null;
}
export interface ShopNowCardSection {
  id: number | null | undefined;
  imageUrl: string | null | undefined;
  altText: string | null | undefined;
  link: string | null | undefined;
  title: string | null | undefined;
  priceRange: string | null | undefined;
  placeholder?: string | null | undefined;
}
